app/channels/program_channel.rb

```diff
@@ -1,11 +1,47 @@
 class ProgramChannel < ApplicationCable::Channel
-  CODE_KEY = "!"
+  periodically :tick, every: 1.seconds
 
   def subscribed
     puts "Subscribed!"
     stream_for room
   end
 
+  def set_mode(message)
+    current_program.update(mode: message["data"])
+    ProgramChannel.broadcast_to(room, { action: :update, data: current_program.tick_view })
+  end
+
+  def set_max_input_mode(message)
+    current_program.update(settings: current_program.settings.merge({
+      max_input_mode: message["data"]
+    }))
+    ProgramChannel.broadcast_to(room, { action: :tick, data: current_program.tick_view })
+  end
+
+  def set_vote_interval(message)
+    current_program.update(settings: current_program.settings.merge({
+      vote_interval: message["data"]
+    }))
+    ProgramChannel.broadcast_to(room, { action: :tick, data: current_program.tick_view })
+  end
+
+  def pause
+    current_program.update(settings: current_program.settings.merge({
+      play_state: "paused" }))
+    ProgramChannel.broadcast_to(room, { action: :tick, data: current_program.tick_view })
+  end
+
+  def resume
+    current_program.update(settings: current_program.settings.merge({
+      play_state: "playing" }))
+    ProgramChannel.broadcast_to(room, { action: :tick, data: current_program.tick_view })
+  end
+
+  def reset_tick
+    current_program.update(tick: 0)
+    ProgramChannel.broadcast_to(room, { action: :tick, data: current_program.tick_view })
+  end
```

Comment:
I've got a few ideas for how best to clean up this repetition, but I wanted to let it sit and do my refactoring in another pass.